spec/models/post_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+
+  describe "DB columns" do
+    it { is_expected.to have_db_column :id }
+    it { is_expected.to have_db_column :caption }
+    it { is_expected.to have_db_column :status }
+    it { is_expected.to have_db_column :category }
+    it { is_expected.to have_db_column :latitude }
+    it { is_expected.to have_db_column :longitude }
+  end
+
+  describe "Validations" do
+    it { is_expected.to validate_presence_of(:caption) }
+    it { is_expected.to validate_presence_of(:status) }
+    it { is_expected.to validate_presence_of(:category) }
+    it { is_expected.to validate_presence_of(:latitude) }
+    it { is_expected.to validate_presence_of(:longitude) }
+    it { is_expected.to validate_length_of(:caption).is_at_most(140) }
+  end
+
+  describe "Factory" do
+    it "should have valid factory" do
+      expect(FactoryBot.create(:post)).to be_valid
+    end
+  end
+
+  describe "Attachment" do
```

Comment:
shouldn't we test for association with `active storage` here?

Reply:
Yes, good catch! We added that now. And also test for the enum attributes `status` and `category`